### PR TITLE
ringbuffer constexpr optimization

### PR DESCRIPTION
**Pull request by yatli**
consolidate put-put_h and get-get_h

Requires `if constexpr` from C++17.

### CODE REVIEW

**Comment by yatli** [comment]
End up with sysex error code 4 🤦‍♂️

**Comment by jmamma** [comment]
MDX_KIT_VERSION has changed to 0x64

File: avr/cores/megacommand/CommonTools/RingBuffer.h [hunk 1]
```diff
@@ -155,18 +153,18 @@ bool CRingBuffer<C, N, T>::put(C *src, T n) volatile {
   }
 
   if constexpr (N != 0) {
```
Comment (jmamma) [inline avr/cores/megacommand/CommonTools/RingBuffer.h]: nice. didn't know about this feature.

File: avr/cores/megacommand/CommonTools/RingBuffer.h [hunk 2]
```diff
@@ -155,18 +153,18 @@ bool CRingBuffer<C, N, T>::put(C *src, T n) volatile {
...
-    memcpy(buf + wr, src, s);
+    memcpy(buf + wr, src, s * sizeof(C));
```
Comment (jmamma) [inline avr/cores/megacommand/CommonTools/RingBuffer.h]: good catch

File: avr/cores/megacommand/CommonTools/RingBuffer.h [hunk 3]
```diff
@@ -92,7 +92,7 @@ CRingBuffer<C, N, T>::CRingBuffer(volatile uint8_t *_ptr) {
 }
 
 template <class C, int N, class T>
-bool CRingBuffer<C, N, T>::put_h_isr(C *src, T n) volatile {
+void CRingBuffer<C, N, T>::put_h_isr(C *src, T n) volatile {
```
Comment (jmamma) [inline avr/cores/megacommand/CommonTools/RingBuffer.h:95]: after all these years, didnt even notice the return type.

**Review by jmamma — APPROVED** [review]

**Comment by jmamma** [comment]
Ah hang on. I'm getting SYSEX ERROR 4 as well.

**Comment by jmamma** [comment]
eh sorry: c46d6d2d9acdccbb5df7c64132b0181112415e9b

**Comment by yatli** [comment]
ok after the merge it's good now

**Pull request merged**